test/test_torch.py

```diff
@@ -17539,45 +17539,33 @@ def is_view_of(self, base, other):
         return True
 
     @onlyOnCPUAndCUDA
-    def test_real_self(self, device):
-        t = torch.ones((5, 5), device=device)
-        s = torch.real(t)
+    @dtypes(torch.float, torch.double, torch.long)
+    def test_real_self(self, device, dtype):
+        t = torch.ones((5, 5), dtype=dtype, device=device)
+        s = t.real
         self.assertTrue(s is t)
 
-        # TODO: update when the imag attribute is implemented
-        self.assertTrue(not hasattr(t, 'real'))
-
-    # TODO: update after torch.real is implemented for complex tensors
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
     @onlyOnCPUAndCUDA
-    def test_real_view(self, device):
-        t = torch.tensor((1 + 1j), device=device)
-        with self.assertRaises(RuntimeError):
-            v = torch.real(t)
-            self.assertTrue(self.is_view_of(t, v))
-
-            v[0] = 0
-            self.assertEqual(t.float()[0], v[0])
-            self.assertTrue(t[0] == complex(0, 1))
+    @dtypes(torch.cfloat, torch.cdouble)
+    def test_real_imag_view(self, device, dtype):
+        t = torch.randn(3, 3, dtype=dtype, device=device)
+        re = t.real
+        exp = torch.from_numpy(t.numpy().real)
+        self.assertEqual(re, exp)
+        assert(t.storage().data_ptr() == re.storage().data_ptr())
```

Comment:
`self.assertEqual(t.storage().data_ptr(), re.storage().data_ptr())`

Reply:
Also: can you not use is_view_of here?

Reply:
`is_view_of` also check the autograd side of things (`other._base is not base`), so will update this test to use is_view_of when the autograd code is modified to handle `.real` and `.imag`

Reply:
OK. If .real and .imag don't work well with autograd, should they be prevented on tensors requiring a gradient? Much like how we don't allow .numpy() on those tensors, either (you have to do tensor.detach().cpu().numpy(), for example).

Reply:
The plan is to fix that behavior in another PR in this stack.
I don't think we need to add this check to remove it in another PR in the same stack that will be merged at the same time anyway.